api/envoy/service/ext_proc/v3/external_processor.proto

```diff
@@ -332,3 +344,16 @@ message BodyMutation {
     bool clear_body = 2;
   }
 }
+
+// This message specifies a new timer duration and the original
+// request direction. If these fields are not configured correctly,
+// the response message will be ignored by the Envoy ext_proc filter.
+message NewProcessingTimeout {
+  // Specifies the new timer duration.
+  // Has to be >= 1ms and <= 10000ms.
+  google.protobuf.Duration message_timeout = 1;
+
+  // Specifies the traffic direction of the received request message.
+  // Has to be INBOUND or OUTBOUND.
```

Comment:
what does this mean when on the response_headers processing, the returned new_timeout direction is on the INBOUND, AND client is actually still sending data? Is that possible?

Reply:
With the new change, this concern can be avoided.